### ISSUE TITLE
[Profile Showcase - Identity] Update `Display name` error messages

### ISSUE THREAD

**Issue by alexandraB99** [status-im/status-desktop#13335]
[Figma](https://www.figma.com/file/ibJOTPlNtIxESwS96vJb06/%F0%9F%91%A4-Profile-%7C-Desktop?type=design&node-id=2460-42200&mode=design&t=jumwFduKExodhRUc-4)

This task should cover the following error messages:
1. `Display Names can’t contain special characters`
<img width="540" alt="Screenshot 2024-02-06 at 14 38 41" src="https://github.com/status-im/status-desktop/assets/47811206/849f7eda-5ad9-44a4-9619-112ab24a6d3c">

2. `Display Names can’t end in “_eth” or “-eth”`
<img width="745" alt="Screenshot 2024-02-06 at 14 39 29" src="https://github.com/status-im/status-desktop/assets/47811206/d084301a-4c1a-4496-a8fb-955a686825fa">

3. `Display Names can’t be longer than 20 characters`
**NOTE: Figma is outdated**
<img width="761" alt="Screenshot 2024-02-06 at 14 39 58" src="https://github.com/status-im/status-desktop/assets/47811206/424de3fa-8da2-428e-a5d8-9abda61a0790">

4. `Display Names must be at least 5 characters long`
<img width="776" alt="Screenshot 2024-02-06 at 14 41 08" src="https://github.com/status-im/status-desktop/assets/47811206/29c0a727-b60c-4514-8261-ee2481375c2e">

5. `This Display Name is already in use in one of your joined communities`
**NOTE: Check if there is a backend for this**
<img width="768" alt="Screenshot 2024-02-06 at 14 41 40" src="https://github.com/status-im/status-desktop/assets/47811206/d1a7ea74-be6c-4cb4-9198-9db0aa59637f">

### Note for development - regarding saving changes

Tooltip tells user which tab the invalid changes are on incase they try and save via a different tab. Only changes to the Identity tab can be invalid (validation for other tabs eg Links is caught within the relevant dialog flows).

**Comment by alexjba** [comment]
@benjthayer We already have an extended set of validators for the `Display name`. Please have a look at the following error messages and let me know if we need to update something. I've merged in this list both the old error messages and the new ones.

1. Usernames starting with whitespace are not allowed _(old error message)_
2.  Usernames ending with whitespace are not allowed _(old error message)_
3. Usernames ending with '.eth' are not allowed _(old error message)_
4. Sorry, the name you have chosen is not allowed, try picking another username _(old error message - here it checks if the display name is an alias)_
5. Display Names can’t contain special characters _(new error message)_
6. Display Names must be at least %n character(s) long _(new error message)_
7. Display Names can’t be longer than %n character(s) _(new error message)_
8. Display Names can’t end in “_eth” or “-eth” _(new error message)_
9. This Display Name is already in use in one of your joined communities _(new error message)_

**Comment by benjthayer** [comment]
Looks good to me, thanks for pulling these historic ones through - I'll ensure they are added to the design @alexandraB99 

A couple of tweaks it might be good to make:

- Standardise the use of `Display Name` (in place of `Username/s`) for consistency.

- For number 4 could you clarify what we are validating there? Is it saying that the entered Display Name is the same as a nickname someone has given that user? I think we probably need to work on the wording of that one based on what is being validated to give the user some insight into why it is not valid so they can edit it accordingly.

- For Display Names ending `.eth`, `_eth`, `-eth` can you remind me what we are validating? Is it:
a) The user cannot enter an ENS name at all OR
b) The user cannot enter an approximation of an ENS name (to limit impersonation) by using `_eth` or `-eth`.
In mobile it is just b) that applies:
https://www.figma.com/file/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?type=design&node-id=11465-89785&mode=design&t=Jtg7LrLnvg2YHrLD-4
and
https://www.figma.com/file/PiN0uwRgZj4L9U4N7xO7Xb/Profile-for-Mobile?type=design&node-id=4409-369325&mode=design&t=HB3w4veluGAcOgPv-4

- We should also add an emojis not allowed one - will add that to the designs

**Comment by alexjba** [comment]
> * For number 4 could you clarify what we are validating there? Is it saying that the entered Display Name is the same as a nickname someone has given that user? I think we probably need to work on the wording of that one based on what is being validated to give the user some insight into why it is not valid so they can edit it accordingly.

@benjthayer Here we're validating the username is not an alias that is usually generated by the app. The format is: two adjectives and an animal. E.g. red blue ant

> * For Display Names ending `.eth`, `_eth`, `-eth` can you remind me what we are validating? Is it:
>   a) The user cannot enter an ENS name at all OR
>   b) The user cannot enter an approximation of an ENS name (to limit impersonation) by using `_eth` or `-eth`.
>   In mobile it is just b) that applies:
>   https://www.figma.com/file/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?type=design&node-id=11465-89785&mode=design&t=Jtg7LrLnvg2YHrLD-4
>   and
>   https://www.figma.com/file/PiN0uwRgZj4L9U4N7xO7Xb/Profile-for-Mobile?type=design&node-id=4409-369325&mode=design&t=HB3w4veluGAcOgPv-4

The user cannot use a display name ending with .eth. This means no ens names being manually configured as display name. The ens names are configured in another settings tab - `ENS usernames`

<img width="1512" alt="Screenshot 2024-02-08 at 12 24 53" src="https://github.com/status-im/status-desktop/assets/47811206/e8925de2-6dc4-46c3-a0a3-f3aed23e34f4">



> * We should also add an emojis not allowed one - will add that to the designs

Detecting emojis is a complication IMO. If the user tries to insert any emoji the validation number 5 would be triggered and we would display this message: `Display Names can’t contain special characters.`. It can be done, of course, but I'm not sure it's needed.

**Comment by benjthayer** [comment]
Thank @alexandraB99 these will probably need some discussion with the design team / John to ensure we're agreed and aligned for the approach on these ones cross-platform. Will add it to our discussion points for our next design sync up.

**Comment by benjthayer** [comment]
Hey @alexjba @alexandraB99 I've updated the Profile editing validation states based on discussions here.

Figma:
https://www.figma.com/file/ibJOTPlNtIxESwS96vJb06/%F0%9F%91%A4-Profile-%7C-Desktop?type=design&node-id=2460-26820&mode=design&t=H5RIhHYaNHdeIzDA-4

A couple of notes:

1) I've combined the validation statements relating to whitespace to a single statement regardless of where the space is:
`Display Names can’t start or end with a space`

2) I've added `.eth` to the ENS validation statement:
`Display Names can’t end in “.eth”, “_eth” or “-eth”`

3) So we give the user a little more information, for validation against the adjective-adjective-animal format, I've adjusted the validation statement to:
`Adjective-animal Display Name formats are not allowed`. This for me, is slightly clearer than `Sorry, the name you have chosen is not allowed, try picking another username` as it clarifies what is wrong and what the user can do to rectify it. WDYT?

--

I've also ensured the nickname character count limit and validation is the same for adding/editing Nicknames via the profile dialog:

Figma:
https://www.figma.com/file/ibJOTPlNtIxESwS96vJb06/%F0%9F%91%A4-Profile-%7C-Desktop?type=design&node-id=2460-202692&mode=design&t=H5RIhHYaNHdeIzDA-4
